packages/lib/services/e2ee/crypto.ts

```diff
@@ -0,0 +1,42 @@
+import { Crypto, HashAlgorithm } from './types';
+import { promisify } from 'util';
+import {
+	getCiphers as nodeGetCiphers, getHashes as nodeGetHashes,
+	randomBytes as nodeRandomBytes,
+	pbkdf2 as nodePbkdf2,
+} from 'crypto';
+
+type NodeDigestNameMap = Record<HashAlgorithm, string>;
```

Comment:
What is it actually? A digest or a hash algorithm? Please pick one name. Also everything is "NodeSomething" in this file but that doesn't mean we should prefix all types and variables. It's implied that it's NodeSomething.

Reply:
> What is it actually? A digest or a hash algorithm? Please pick one name.

They are almost identical in this context. For `node:crypto` and `react-native-quick-crypto`, the argument name in `pbkdf2()` is `digest` but the type of it is `HashAlgorithm`. considering the digest is the result of a hash algorithm I will use `hashxx` for all cases except the argument name in `pbkdf2Raw()` (to match the argument name in `node:crypto` and `react-native-quick-crypto`).

> Also everything is "NodeSomething" in this file but that doesn't mean we should prefix all types and variables. It's implied that it's NodeSomething.

I use the prefix `node` to indicate the function is from the `node:crypto` rather than my own implementation. If I remove the prefix, the name `getCiphers` could refers to the one in `node:crypto` and the one in my interface `Crypto`. In the upcoming draft I didn't add the `node` prefix to functions like `createCipheriv` because I won't implement a wrapper of it in the interface `Crypto`.
Is this acceptable?

Reply:
yes

Reply:
Finished in commit e4eb58c21df17d51000610cae52594835c9eafd6.